File: coupon/coupon-api/src/main/java/dev/be/coupon/api/coupon/presentation/dto/CouponCreateResponse.java
package dev.be.coupon.api.coupon.presentation.dto;

import dev.be.coupon.api.coupon.application.dto.CouponCreateResult;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public record CouponCreateResponse(
        UUID id,
        String couponName,
        String couponType,
        String couponDiscountType,
        BigDecimal couponDiscountValue,
        int totalQuantity,
        String couponStatus,
        LocalDateTime expiredAt
) {
    public static CouponCreateResponse from(final CouponCreateResult result) {
        return new CouponCreateResponse(
                result.id(),
                result.couponName(),
                result.couponType(),
                result.couponDiscountType(),
                result.couponDiscountValue(),
                result.totalQuantity(),
                result.couponStatus(),
                result.expiredAt()
        );
    }
}

File: coupon/coupon-api/src/main/kotlin/dev/be/coupon/api/coupon/presentation/dto/CouponCreateResponse.kt
package dev.be.coupon.api.coupon.presentation.dto

import dev.be.coupon.api.coupon.application.dto.CouponCreateResult
import java.time.LocalDateTime
import java.util.*

data class CouponCreateResponse(
    val id: UUID,
    val name: String,
    val type: String,
    val totalQuantity: Int,
    val status: String,
    val validFrom: LocalDateTime,
    val validUntil: LocalDateTime
) {
    companion object {
        fun from(result: CouponCreateResult): CouponCreateResponse {
            return CouponCreateResponse(
                id = result.id,
                name = result.name,
                type = result.type,
                totalQuantity = result.totalQuantity,
                status = result.status,
                validFrom = result.validFrom,
                validUntil = result.validUntil
            )
        }
    }
}
